tests/integration/individual_response/test_individual_response.py

```diff
@@ -0,0 +1,375 @@
+from tests.integration.integration_test_case import IntegrationTestCase
+
+
+class IndividualResponseTestCase(IntegrationTestCase):
+    def setUp(self):
+        super().setUp()
+        self.launchSurvey("test_individual_response")
+
+    @property
+    def individual_section_link(self):
+        return self.getHtmlSoup().findAll(
+            "a", {"data-qa": "summary-actions-summary-button"}
+        )[1]["href"]
+
+    @property
+    def individual_response_link(self):
+        return (
+            self.getHtmlSoup()
+            .find("p", {"data-qa": "individual-response-url"})
+            .find_next()["href"]
+        )
+
+    @property
+    def previous_link(self):
+        return self.getHtmlSoup().find("a", {"data-ga-action": "Previous link click"})[
+            "href"
+        ]
+
+    @property
+    def sign_out_link(self):
+        return self.getHtmlSoup().find("a", {"name": "btn-save-sign-out"})["href"]
+
+    def _add_no_household_members(self):
+        self.get("questionnaire/primary-person-list-collector/")
+        self.post({"you-live-here": "No"})
+        self.post({"anyone-else": "No"})
+        self.get("questionnaire/")
+
+    def _add_primary(self):
+        self.get("questionnaire/primary-person-list-collector/")
+        self.post({"you-live-here": "Yes"})
+        self.post({"first-name": "Marie", "last-name": "Day"})
+        self.post({"anyone-else": "No"})
+        self.get("questionnaire/")
+
+    def _add_primary_and_household(self):
+        self.get("questionnaire/primary-person-list-collector/")
+        self.post({"you-live-here": "Yes"})
+        self.post({"first-name": "Marie", "last-name": "Day"})
+        self.post({"anyone-else": "Yes"})
+        self.post({"first-name": "John", "last-name": "Doe"})
+        self.post({"anyone-else": "No"})
+        self.get("questionnaire/")
+
+    def _add_household_no_primary(self):
+        self.get("questionnaire/primary-person-list-collector/")
+        self.post({"you-live-here": "No"})
+        self.post({"anyone-else": "Yes"})
+        self.post({"first-name": "Marie", "last-name": "Day"})
+        self.post({"anyone-else": "No"})
+        self.get("questionnaire/")
+
+    def _request_individual_response(self):
+        self._add_household_no_primary()
+        self.post()
+        self.get(self.individual_response_link)
+        self.post()
+        self.post({"individual-response-how-answer": "Post"})
+        self.post(
+            {
+                "individual-response-post-confirm-answer": "Yes, send the access code by post"
+            }
+        )
+        self.post()
+
+
+class TestIndividualResponseErrorStatus(IndividualResponseTestCase):
+    def test_401_how_redirects_signed_out(self):
```

Comment:
`test_ir_is_not_accessible_without_session` or `test_ir_is_not_accessible_after_signout` or `test_ir_raises_401_without_session`?. Something along those lines as the current one wasn't clear to me.

Reply:
I've gone with `test_ir_raises_401_without_session`